test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/Search/DateSearchTests.cs

```diff
@@ -150,5 +152,25 @@ public async Task GivenADateTimeSearchParam_WhenSearched_ThenCorrectBundleShould
 
             ValidateBundle(bundle, expected);
         }
+
+        [Theory]
+        [InlineData("***")]
+        [InlineData("!")]
+        public async Task GivenAnInvalidDateTimeSearchParam_WhenSearched_ThenExceptionShouldBeThrown(string queryValue)
+        {
+            var fhirException = await Assert.ThrowsAsync<FhirException>(async () => await Client.SearchAsync(ResourceType.Patient, $"birthdate={queryValue}"));
+
+            Assert.Equal(HttpStatusCode.BadRequest, fhirException.StatusCode);
+        }
+
+        [Theory]
+        [InlineData("1973-02-30")]
+        [InlineData("1973-02-28T01:01:09.999999999999999999")]
+        public async Task GivenAnOutOfRangeDateTimeSearchParam_WhenSearched_ThenExceptionShouldBeThrown(string queryValue)
+        {
+            var fhirException = await Assert.ThrowsAsync<FhirException>(async () => await Client.SearchAsync(ResourceType.Patient, $"birthdate={queryValue}"));
+
+            Assert.Equal(HttpStatusCode.BadRequest, fhirException.StatusCode);
+        }
```

Comment:
These tests ensure that `HttpStatusCode.InternalServerError` is not returned. I'm considering increasing the coverage of these tests once #25 and #911 are addressed.